nest-bot/src/os/os_functions.ts

```diff
@@ -1,14 +1,8 @@
 import { readFile, writeFile } from "fs/promises";
-import { exec, ExecException } from "child_process";
+import { exec, ExecException, execSync } from "child_process";
 
 export async function add_root_caddyfile_config(username: string) {
-  const templateConfig = (
-    await readFile("./src/os/templates/root_caddyfile_config.txt")
-  ).toString("utf8");
-
-  const newConfig = templateConfig.replace(/<username>/g, username);
-
-  await writeFile("/etc/caddy/Caddyfile", newConfig, { flag: "a" });
+  execSync(`nest caddy add ${username}.hackclub.app --user ${username}`)
```

Comment:
Nest Bot is run as `nest-internal` user, so if the `--user` flag is checking for `root` then you will need to sudo

Reply:
Fixed with latest commit